Benchmarks/Benchmarks/NIOCoreBenchmarks/Benchmarks.swift

```diff
@@ -21,6 +21,11 @@ let benchmarks = {
         .mallocCountTotal
     ]
 
+    let leakMetrics: [BenchmarkMetric] = [
+        .mallocCountTotal,
+        .memoryLeaked,
+    ]
```

Comment:
I wanted to add one comment here. Should we add leaks to all of our benchmarks. Seems like we had that before in the old school alloc tests.

Reply:
_Maybe_. Unfortunately it seems like there isn't a simple "detect leaks" lever to pull here. Even in a case where I manufactured a leak I had to tweak the configuration carefully to have it detected. We could copy that configuration and hope it would detect _any_ leak but doing so would slow down the benchmarks significantly. I'm planning on investigating jemalloc in more detail to see if we can do anything to make leak detection more robust.